main/tests.py

```diff
@@ -72,8 +74,66 @@ def test_unlogged_user_access_to_homepage(self, client):
         assert response.status_code == 302
         assert response['Location'] == '/login/?next=/homepage/'
 
-    def test_dog_owner_homepage_is_visible_for_dog_owner(self, client, create_dog_owner_user):
+
+@pytest.mark.django_db
+class TestDogOwnerHomePageView:
+    def test_dog_owner_homepage_present_all_daycares(self, client, create_dog_owner_user):
         client.force_login(user=create_dog_owner_user.user)
         response = client.get("/homepage/")
         assert response.status_code == 200
-        assert list(response.context['daycares']) == list(DayCare.objects.all())
+        assert set(response.context['day_care_queryset']) == set(DayCare.objects.all())
+
+    def test_daycare_not_appears_in_search_after_reduce_capacity_on_dates_to_0(self, client,
+                                                                               create_dog_owner_user,
+                                                                               create_daycare_user):
+        search_form = {'area': "",
+                       'city': "",
+                       'price_per_day': 100,
+                       'name': "",
+                       'start_date': "2022-05-03",
+                       'end_date': "2022-05-08",
+                       }
+        daycare_user = create_daycare_user
+        client.force_login(user=create_dog_owner_user.user)
+        response = client.post('/homepage/', search_form, follow=True)
+        day_care_queryset = response.context['day_care_queryset']
+        assert daycare_user in day_care_queryset
```

Comment:
It would make the test look nicer if the assertion would be separated with newline from the test setup logic (all over this file).